src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2049,6 +2049,18 @@ private TypeParameterConstraintSyntax ParseTypeParameterConstraint()
                 case SyntaxKind.DefaultKeyword:
                     var defaultToken = this.EatToken();
                     return CheckFeatureAvailability(_syntaxFactory.DefaultConstraint(defaultToken), MessageID.IDS_FeatureDefaultTypeParameterConstraint);
+                case SyntaxKind.EnumKeyword:
+                    {
+                        var missingType = this.AddError(this.CreateMissingIdentifierName(), ErrorCode.ERR_NoEnumConstraint);
+                        missingType = AddTrailingSkippedSyntax(missingType, this.EatToken());
+                        return _syntaxFactory.TypeConstraint(missingType);
```

Comment:
i still think we should make this a struct-constraint (to help with later stuff).  but it's nbd.

Reply:
It's tempting, but I'd rather not give too much meaning to a bad token. It could be a legitimate misspelling (from `Anum`?).

Reply:
i think that seems highly ulikely.  `enum` is a strong c# keyword, and i personally think it's totally clear what hte user is trying to say when they add that constraint.  Indeed, it's a constraint we've even considered, we just didn't do it since it falls out from `where T : struct, System.Enum` :)